exec/table_exec_command.go

```diff
@@ -138,45 +145,83 @@ func listExecCommand(ctx context.Context, d *plugin.QueryData, h *plugin.Hydrate
-	plugin.Logger(ctx).Debug("listExecCommand", "ctx_done", "finished")
+	plugin.Logger(ctx).Debug("listExecCommand", "ctx_done", "adding row...")
 
-	return nil, nil
+	// If the command failed, return the stderr output
+	if result.ExitCode != 0 {
+		d.StreamListItem(ctx, commandResult{StderrOutput: stdout, ExitCode: result.ExitCode})
+		return nil, nil
+	}
 
+	d.StreamListItem(ctx, commandResult{StdoutOutput: stdout, ExitCode: result.ExitCode})
+	return nil, nil
 }
 
-func listLocalCommandResult(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+func listLocalCommand(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
 
 	cmd, err := prepareCommand(ctx, d, h)
 	if err != nil {
-		plugin.Logger(ctx).Error("listLocalCommandResult", "command_error", err)
+		plugin.Logger(ctx).Error("listLocalCommand", "prepareCommand", "command_error", err)
 		return nil, err
 	}
 
 	if cmd == nil {
 		// Empty command returns zero rows
-		plugin.Logger(ctx).Debug("listLocalCommandResult", "cmd", cmd)
-		return nil, nil
+		plugin.Logger(ctx).Debug("listLocalCommand", "cmd", cmd)
+		return nil, err
+	}
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		plugin.Logger(ctx).Error("listLocalCommand", "error reading stdout", err)
+		return nil, err
 	}
 
-	output, err := cmd.CombinedOutput()
+	stderr, err := cmd.StderrPipe()
 	if err != nil {
+		plugin.Logger(ctx).Error("listLocalCommand", "error reading stderr", err)
+		return nil, err
+	}
+
+	if err := cmd.Start(); err != nil {
+		plugin.Logger(ctx).Error("listLocalCommand", "error starting command", err)
+		return nil, err
+	}
+
+	// Create slices to store standard output and standard error lines
+	var stdoutLines []string
+	var stderrLines []string
+
+	// Read standard output and standard error concurrently
+	go func() {
+		scanner := bufio.NewScanner(stdout)
+		for scanner.Scan() {
+			line := scanner.Text()
+			stdoutLines = append(stdoutLines, line)
+		}
+	}()
+
+	go func() {
+		scanner := bufio.NewScanner(stderr)
+		for scanner.Scan() {
+			line := scanner.Text()
+			stderrLines = append(stderrLines, line)
+		}
+	}()
+
+	if err := cmd.Wait(); err != nil {
 		// Log the error, but don't fail. The command error output will be captured
 		// and returned to the user.
-		plugin.Logger(ctx).Error("listLocalCommandResult", "command_error", err)
+		plugin.Logger(ctx).Error("listLocalCommand", "cmd.Wait", "command_error", err)
 	}
 
-	outputStr := string(output)
-
-	// NOTE - I considered stripping the final newline. The output string looks
-	// weird with the final newline (appears like an extra newline) in SQL
-	// results, so is confusing. But removing the newline is altering the output,
-	// so on balance better to leave it accurate. I'm leaving this code here as a
-	// warning to anyone who is tempted to strip the newline in the future.
-	// outputStr := strings.TrimSuffix(outputStr, "\n")
+	stdoutString := strings.Join(stdoutLines, "\n")
```

Comment:
Instead of trimming, we're now using newlines in join, what effect does this have on query results?

Reply:
fixed